app/controllers/main/routes.py

```diff
@@ -170,3 +172,20 @@ def searchUser(query):
 @main_bp.route('/contributors',methods = ['GET'])
 def contributors():
     return render_template("/contributors.html")
+
+
+
+@main_bp.route('/manageservicelearning', methods = ['GET'])
+def getAllCourseIntructors():
+    """
+    This function selects all the Intructors Name and the previous courses
+    """
+    users = User.select().where(User.isFaculty)
+    courseInstructors = CourseInstructor.select()
+    course_dict = {}
+
+    for i in courseInstructors:
+        course_dict.setdefault(i.user.firstName + " " + i.user.lastName, []).append(i.course.courseName)
```

Comment:
You are still using first + " " + last as a key. You should use the user as the key. Right now you are doing formatting in the controller instead of the view.

Reply:
Also, you should have a getAllCourseInstructors method to get this data structure that you can test.

Reply:
Created a getCourseDict method in logic file manageSLFaculty.py